Pidgin/Parser.Enum.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Pidgin
+{
+    public static partial class Parser
+    {
+        /// <summary>
+        /// Creates a parser that parses and returns one of enum values
+        /// </summary>
+        /// <param name="ignoreCase">Flag, true: parser  values in a case insensitive manner, false: case sensitive</param>
+        /// <typeparam name="TEnum">Enum type</typeparam>
+        /// <returns>A parser that parses and returns one of enum values</returns>
+        public static Parser<char, TEnum> ParseEnum<TEnum>(bool ignoreCase = false)
```

Comment:
I think it’s probably more likely, in compilers for languages, that users would want to force the keywords to be lowercase, rather than case insensitive.